### PR TITLE
Fix #961 -- Move view detail permissions into project user policy

### LINKED ISSUE #961: Permissions issue: Public user role within an organization
### Steps to reproduce the error

Based off this user's permissions (which I setup locally):
screenshot 2016-12-01 08 13 12
 
The user should have public user rights to the H4H Test Project.  Generally this user should have the  ability to view the project overview page with the locations on the map.  This user should not be able to view location details, relationships, parties, resources. 

### Actual behavior

I'm able to view all the location details, relationships, parties and resources through the location popup on the overview map.  Following PR #942, the sidebar is removed within the project for public users. 

screenshot 2016-12-01 10 42 56

### Expected behavior

I should get a message that I don't have permission to view the location data (and ideally an option to request permission.)

### PR DESCRIPTION
**Pull request by oliverroick**
### Proposed changes in this pull request

- Fixes #961: Moving `view` permissions for all entities into the project user policy description. Organisation members will not be able to view details for locations, relationships, parties and resources for any of the organisation's projects; unless they are a member of the project.
- Change permission handling of `PartiesList` to make it consistent with locations listing: All parties are listed, not taking into account whether the user can view details or not. 

### When should this PR be merged

Anytime

### Risks

None

### Follow up actions

Permissions policies need to be reloaded when deploying (/cc @amplifi)

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description. 
- [ ] **Is the PR labeled correctly?** It should have the `migration` label if a new migration is added. 
**Is the risk level assessment sufficient?** The risks section should contain all risks that might be introduced with the PR and which actions we need to take to mitigate these risks. Possible risks are database migrations, new libraries that need to be installed or changes to deployment scripts. 

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled. 

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests. 
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/). 

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components. 
- [ ] **If this is a bug fix, are tests for the issue in place**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes must be documented in the [Cadasta Platform Documentation](https://github.com/Cadasta/cadasta-docs).
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented in the [API docs](https://github.com/Cadasta/api-docs).
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: cadasta/config/permissions/project-user.json [hunk 1]
```diff
@@ -4,5 +4,35 @@
     // additional permissions over those given to all users.  This may
     // change in the future.  In particular, project users may be
     // permitted access to projects that are normally private.
```
Comment (seav) [inline cadasta/config/permissions/project-user.json]: This comment is no longer correct ("no additional permissions").

File: cadasta/party/tests/test_views_default.py [hunk 2]
```diff
@@ -84,8 +84,27 @@ def test_get_from_non_existent_project(self):
     def test_get_with_unauthorized_user(self):
         user = UserFactory.create()
         response = self.request(user=user)
+        assert response.status_code == 302
```
Comment (seav) [inline cadasta/party/tests/test_views_default.py]: Please add another assertion for the redirect URL.

Comment (oliverroick) [inline cadasta/party/tests/test_views_default.py]: This isn't necessary.

The redirect URL is based on the referrer and it is defined in `core.mixins.PermissionRequiredMixin.handle_no_permission`, which has appropriate tests.

**Comment by seav** [comment]
The updated changes are OK.

**Pull request merged**